src/parser/AttributeValueParser.java

```diff
@@ -0,0 +1,36 @@
+package parser;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class AttributeValueParser<T> {
+    private static String idRegex = " Id :(.)+";
+    private static String colorRegex = "Color :(.)+";
+    private static String defaultMessage = "Value not found!";
+    private static final Pattern ID_PATTERN = Pattern.compile(idRegex);
+    private static final Pattern COLOR_PATTERN = Pattern.compile(colorRegex);
+    private static Matcher matcher;
```

Comment:
Put it inside method. You don't need to use it in multiple methods due to having single one.